### PR TITLE
Added shortcut for route

### PR DESCRIPTION
**Pull request by surajnarwade**

### CODE REVIEW

**Comment by surajnarwade** [comment]
@kadel @cdrage review needed :)

**Comment by kadel** [comment]
test are failing :-(

**Comment by surajnarwade** [comment]
@kadel tests are failing because `routeEndpoint` and `ingressEndpoint` is not present in JSON schema

**Comment by kadel** [comment]
hmm :-(
this needs to be fixed either in this PR, or before this PR gets merged

**Comment by surajnarwade** [comment]
@kadel , added tests

File: docs/file-reference.md [hunk 1]
```diff
@@ -727,11 +727,12 @@ services:
...
-| Field        | Type            | Required | Description                       |
-|--------------|-----------------|----------|-----------------------------------|
-| name         | string          | yes      | The name of the service.          |
-| endpoint     | string          | no       | The endpoint of the service.      |
-| portMappings | array of "port" | no       | Array of ports. Ex. `80:8080/tcp` |
+| Field           | Type            | Required | Description                       |
+|-----------------|-----------------|----------|-----------------------------------|
+| name            | string          | yes      | The name of the service.          |
+| ingressEndpoint | string          | no       | The endpoint of the service.      |
+| routeEndpoint   | string or bool  | no       | The route endpoint of service
```
Comment (cdrage) [inline docs/file-reference.md]: missing a `|` at the end

File: pkg/spec/resources.go [hunk 2]
```diff
@@ -417,7 +417,38 @@ func (app *App) createServices() ([]runtime.Object, error) {
...
+							Name: svc.Name,
+						},
+					},
+				}
+
+				switch port.RouteEndpoint {
+				case "true":
+					// OpenShift will take care of route
+					svcs = append(svcs, endpointRoute)
+				case "false":
```
Comment (cdrage) [inline pkg/spec/resources.go]: remove this case, don't need it if were just doing a debug

File: pkg/spec/resources.go [hunk 3]
```diff
@@ -417,7 +417,38 @@ func (app *App) createServices() ([]runtime.Object, error) {
...
+					// Do not create route
+					log.Debug("Don't create route")
+				default:
+					// Route URL is given by user
+					endpointRoute.Spec.Host = port.RouteEndpoint
+					svcs = append(svcs, endpointRoute)
+
```
Comment (cdrage) [inline pkg/spec/resources.go]: i believe you can shorten this to an if statement instead.. ex:

```go
if port.RouteEndpoint {
 svcs = append(svcs, endpointRoute)
} else {
 endpointRoute.Spec.Host = port.RouteEndpoint
 svcs = append(svcs, endpointRoute)
}
```

**Review by cdrage — CHANGES_REQUESTED** [review]

File: pkg/spec/resources.go (hunk 3 above)
Comment (surajnarwade) [reply pkg/spec/resources.go]: @cdrage , since port.RouteEndpoint is string, I can't do that

File: docs/examples/route-endpoint/httpd-false.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+- containers:
+  - image: bitnami/nginx
+
+services:
+- name: httpd
+  type: NodePort
+  ports:
+    - port: 8080
+      targetPort: 8080
+      routeEndpoint: false
```
Comment (containscafeine) [inline docs/examples/route-endpoint/httpd-false.yml]: do we need an example with false?

File: docs/file-reference.md [hunk 5]
```diff
@@ -778,6 +779,26 @@ of `service`.
...
+OR
+
+```yaml
+routeEndpoint: true
+```
+| Type           | Required | Description                  |
+|----------------|----------|------------------------------|
+| string or bool | no       | The endpoint of the service. |
+
+This is an added field in the Service port, which if specified an `route`
```
Comment (containscafeine) [inline docs/file-reference.md]: s/an/a

**Review by concaf — APPROVED** [review]

**Comment by surajnarwade** [comment]
@containscafeine , @kadel can we merge this ?
